src/control/agent/main.go

```diff
@@ -46,19 +56,68 @@ func main() {
 	}
 }
 
+// applyCmdLineOverrides will overwrite Configuration values with any non empty
+// data provided, usually from the commandline.
+func applyCmdLineOverrides(c *client.Configuration, RuntimeDir string, LogFile string) error {
```

Comment:
why don't you supply the cliOpts struct as a function parameter, then you don't have to change the function signature when adding members to cliOpts